static/js/stream_edit.js

```diff
@@ -446,6 +446,8 @@ function change_stream_privacy(e) {
     }
 
     settings_ui.do_settings_change(channel.patch, url, data, status_element);
+
+    stream_settings_ui.update_privacy_icon({invite_only, is_web_public, color: sub.color});
```

Comment:
This should not be called here as then the icon would be changed even if the server returns error. You can try it by changing a default stream to be private. This should be called in the relevant function in `stream_ui_updates.js` which is executed after receiving update event from the server.